### PR TITLE
Add new resource service_private_endpoints_attachment

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/ClickHouse/terraform-provider-clickhouse/issues/137

Adding a new resource to be able to attach a private endpoint to a clickhouse service.

### CODE REVIEW

File: pkg/resource/models/service_private_endpoints_attachment.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package models
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type ServicePrivateEndpointsAttachment struct {
+	PrivateEndpointIds types.List   `tfsdk:"private_endpoint_ids"`
```
Comment (nikita-vanyasin) [inline pkg/resource/models/service_private_endpoints_attachment.go:8]:
```suggestion
	PrivateEndpointIDs types.List   `tfsdk:"private_endpoint_ids"`
```

This is kinda golang way to use abbreviations in var/fields names

File: pkg/resource/service_private_endpoints_attachment.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+	plan.PrivateEndpointIds.ElementsAs(ctx, &servicePrivateEndpointIds, false)
+	for _, item := range servicePrivateEndpointIds {
+		service.PrivateEndpointIds.Add = append(service.PrivateEndpointIds.Add, item.ValueString())
+	}
+
+	_, err := r.client.UpdateService(plan.ServiceId.ValueString(), service)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Registering ClickHouse Organization Private Endpoint IDs",
+			"Could not update organization private endpoint IDs, unexpected error: "+err.Error(),
```
Comment (egor-krv) [inline pkg/resource/service_private_endpoints_attachment.go:118]: Maybe add endpoint IDs into the message.

File: pkg/resource/service_private_endpoints_attachment.go [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+	// Get latest service value from ClickHouse OpenAPI
+	service, err := r.client.GetService(state.ServiceId.ValueString())
+	if api.IsNotFound(err) {
+		// Service not found, hence attachment cannot exist as well.
+		resp.State.RemoveResource(ctx)
+		return
+	} else if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading ClickHouse Service",
+			"Could not read ClickHouse service edpoints service id"+state.ServiceId.ValueString()+": "+err.Error(),
```
Comment (nikita-vanyasin) [inline pkg/resource/service_private_endpoints_attachment.go:147]:
```suggestion
			"Could not read ClickHouse service private endpoints service id"+state.ServiceId.ValueString()+": "+err.Error(),
```

**Review by nikita-vanyasin — APPROVED** [review]

**Review by egor-krv — APPROVED** [review]

File: pkg/resource/models/service_private_endpoints_attachment.go (hunk 1 above)
Comment (whites11) [reply pkg/resource/models/service_private_endpoints_attachment.go]: done

**Pull request merged**